File: ZipFlash_debugBuild/app/src/main/java/com/zipflash/mrp/models/AppInfo.java
package com.zipflash.mrp.models;

import android.graphics.drawable.Drawable;

public class AppInfo {
    private String name;
    private String packageName;
    private String activityName;   // null = app entry, not null = activity entry
    private Drawable icon;

    public AppInfo(String name, String packageName, Drawable icon) {
        this.name = name;
        this.packageName = packageName;
        this.icon = icon;
        this.activityName = null;
    }

    public AppInfo(String name, String packageName, String activityName, Drawable icon) {
        this.name = name;
        this.packageName = packageName;
        this.activityName = activityName;
        this.icon = icon;
    }

    public String getName() { return name; }
    public String getPackageName() { return packageName; }
    public String getActivityName() { return activityName; }
    public Drawable getIcon() { return icon; }
    public boolean isActivity() { return activityName != null; }
}

File: ZipFlash/app/src/main/java/com/zipflash/mrp/models/AppInfo.kt
package com.zipflash.mrp.models

import android.graphics.drawable.Drawable

data class AppInfo(
    val name: String,
    val packageName: String,
    val activityName: String?,
    val icon: Drawable
) {
    constructor(name: String, packageName: String, icon: Drawable) : this(name, packageName, null, icon)

    fun isActivity(): Boolean = activityName != null
}
